data/src/main/java/org/apache/iceberg/data/orc/GenericOrcReaders.java

```diff
@@ -0,0 +1,272 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.data.orc;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.iceberg.data.DateTimeUtil;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.orc.OrcValueReader;
+import org.apache.iceberg.orc.OrcValueReaders;
+import org.apache.iceberg.types.Types;
+import org.apache.orc.storage.ql.exec.vector.BytesColumnVector;
+import org.apache.orc.storage.ql.exec.vector.ColumnVector;
+import org.apache.orc.storage.ql.exec.vector.DecimalColumnVector;
+import org.apache.orc.storage.ql.exec.vector.ListColumnVector;
+import org.apache.orc.storage.ql.exec.vector.LongColumnVector;
+import org.apache.orc.storage.ql.exec.vector.MapColumnVector;
+import org.apache.orc.storage.ql.exec.vector.TimestampColumnVector;
+
+
+public class GenericOrcReaders {
+  private static final OffsetDateTime EPOCH = Instant.ofEpochSecond(0).atOffset(ZoneOffset.UTC);
+  private static final LocalDate EPOCH_DAY = EPOCH.toLocalDate();
+
+  private GenericOrcReaders() {
+  }
+
+  public static OrcValueReader<Record> struct(
+      List<OrcValueReader<?>> readers, Types.StructType struct, Map<Integer, ?> idToConstant) {
+    return new StructReader(readers, struct, idToConstant);
+  }
+
+  public static OrcValueReader<List<?>> array(OrcValueReader<?> elementReader) {
+    return new ListReader(elementReader);
+  }
+
+  public static OrcValueReader<Map<?, ?>> map(OrcValueReader<?> keyReader, OrcValueReader<?> valueReader) {
+    return new MapReader(keyReader, valueReader);
+  }
+
+  public static OrcValueReader<OffsetDateTime> timestampTzs() {
+    return TimestampTzReader.INSTANCE;
+  }
+
+  public static OrcValueReader<BigDecimal> decimals() {
+    return DecimalReader.INSTANCE;
+  }
+
+  public static OrcValueReader<String> strings() {
+    return StringReader.INSTANCE;
+  }
+
+  public static OrcValueReader<UUID> uuid() {
+    return UUIDReader.INSTANCE;
+  }
+
+  public static OrcValueReader<ByteBuffer> bytes() {
```

Comment:
Rename to `bytebuffers()` and `ByteBufferReader`? Less confusion between `OrcValueReader.bytes()` and this

Reply:
I would prefer to keep it as bytes. bytes[] vs ByteBuffer seems the datatypes which Spark and Generic Reader understand. The high level idea is still returning bytes in some form.

Reply:
I think the current name is okay, too.